examples/p4est_3d_dgsem/elixir_navierstokes_viscous_shock_dirichlet_bc.jl

```diff
@@ -0,0 +1,169 @@
+using OrdinaryDiffEq
+using Trixi
+
+# This is the classic 1D viscous shock wave problem with analytical solution 
+# for a special value of the Prandtl number.
+# The original references are:
+#
+# - R. Becker (1922)
+#   Stoßwelle und Detonation.
+#   [DOI: 10.1007/BF01329605](https://doi.org/10.1007/BF01329605)
+#
+#   English translations:
+#   Impact waves and detonation. Part I.
+#   https://ntrs.nasa.gov/api/citations/19930090862/downloads/19930090862.pdf
+#   Impact waves and detonation. Part II.
+#   https://ntrs.nasa.gov/api/citations/19930090863/downloads/19930090863.pdf
+#
+# - M. Morduchow, P. A. Libby (1949)
+#   On a Complete Solution of the One-Dimensional Flow Equations 
+#   of a Viscous, Head-Conducting, Compressible Gas
+#   [DOI: 10.2514/8.11882](https://doi.org/10.2514/8.11882)
+#
+#
+# The particular problem considered here is described in
+# - L. G. Margolin, J. M. Reisner, P. M. Jordan (2017) 
+#   Entropy in self-similar shock profiles
+#   [DOI: 10.1016/j.ijnonlinmec.2017.07.003](https://doi.org/10.1016/j.ijnonlinmec.2017.07.003)
+
+### Fixed parameters ###
+
+# Special value for which nonlinear solver can be omitted
+# Corresponds essentially to fixing the Mach number
+alpha = 0.5
+# We want kappa = cp * mu = mu_bar to ensure constant enthalpy
+prandtl_number() = 3 / 4
+
+### Free choices: ###
+gamma() = 5 / 3
```

Comment:
This doesn't need to be a function, does it? Same with several of the functions below - the only things that need to be functions should be Prandtl, right? 

Does this need to be a function to ensure type stability/no allocations?

Reply:
So they can be variables, but in that case calling the initial condition (e.g. via the boundary conditions) causes allocations, since these are global variables. To avoid that, these are declared as functions.

Reply:
But maybe this is a good chance to show that this behaviour can also achieved by setting those global variables as `const`, I will do that.